core/include/scipp/core/element/special_values.h

```diff
@@ -13,23 +13,23 @@
 namespace scipp::core::element {
 
 constexpr auto isnan =
-    overloaded{arg_list<double, float>,
+    overloaded{arg_list<double, float>, transform_flags::no_out_variance,
```

Comment:
Why is this flag required? The function has no output, and even if it does it is `bool`, which can never have variances.

Reply:
Not sure what you mean The function has no output? It returns a variable. This flag is required for it to operate with variables containing variances otherwise you run into this https://github.com/scipp/scipp/blob/master/variable/include/scipp/variable/data_model.h#L67
In current master
```python
import scipp as sc
import numpy as np
a = sc.Variable(['x'], values=np.arange(3.0), variances=np.arange(3.0))
sc.isnan(a)
```

```RuntimeError: This data type cannot have variances.```

Reply:
I just checked in `transform`, and it looks like you are right. This can and should be fixed there though, I think (I thought it did already)?

https://github.com/scipp/scipp/blob/42dcf7f7a4cc025b9dddaf0fc18beaa30c4c5a6d/variable/include/scipp/variable/transform.h#L255-L258

Above should contain something like `&& canHaveVariances<Out>()`?

Reply:
Ok. I'll apply the fix here and put some tests around it.